src/block/Nylium.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\block\utils\BlockEventHelper;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+use pocketmine\utils\Random;
+use function count;
+use function mt_rand;
+
+class Nylium extends Opaque{
+
+	public function getDropsForCompatibleTool(Item $item) : array{
+		return [
+			VanillaBlocks::NETHERRACK()->asItem()
+		];
+	}
+
+	public function isAffectedBySilkTouch() : bool{
+		return true;
+	}
+
+	public function ticksRandomly() : bool{
+		return true;
+	}
+
+	public function onRandomTick() : void{
+		$blockAbove = $this->getSide(Facing::UP);
+		if(!$blockAbove->isTransparent() && $blockAbove->getTypeId() === BlockTypeIds::SNOW_LAYER){
+			//nylium dies
+			BlockEventHelper::spread($this, VanillaBlocks::NETHERRACK(), $this);
```

Comment:
Is this really a block spread ?

Reply:
> Similar to [grass blocks](https://minecraft.wiki/w/Grass_block) and [mycelium](https://minecraft.wiki/w/Mycelium), nylium decays (based on [random ticks](https://minecraft.wiki/w/Random_tick)) into netherrack

Should I use BlockEventHelper::die()?

Reply:
Die isn't the right event, its use litteral for coral etc ...
Its not really a spread or a form. I don't see any created event which can handle it.

Reply:
Should I just replace it without a event?

Reply:
We need this event, but in other way, Nylium is dying, however it can be confusing

Reply:
Should I add a function called decay()?

Reply:
Okay, so, if we reproduce the logics of grass or mycelium. We need to call a spread event.

Reply:
Alright xD